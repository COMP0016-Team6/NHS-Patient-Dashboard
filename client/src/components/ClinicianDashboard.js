import React, { useEffect, useState } from "react";
import {Link} from "react-router-dom";
import { Layout, Menu, Breadcrumb, Row, Col, Button, Avatar, Popover, Divider } from "antd";
import {
  AreaChartOutlined,
  BellOutlined,
  ContactsOutlined,
  UserOutlined,
} from "@ant-design/icons";
import SearchableDashboardTable from "./DashboardTable";

const { Header, Content, Sider } = Layout;

const ClinicianDashboard = ({ logout }) => {
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [patients, setPatients] = useState([]);

  useEffect(() => {
    let cancelled = false;

    const getProfile = async () => {
      try {
        const res = await fetch("http://localhost:5000/dashboard/", {
          method: "POST",
          headers: { jwt_token: localStorage.token }
        });
        const parseData = await res.json();
        if (!cancelled) {
          setName(parseData.user_name);
          setEmail(parseData.user_email);
        }
        
      } catch (err) {
        console.error(err.message);
      }
    };
    getProfile();
    return () => cancelled = true; 
  }, []);

  useEffect(() => {
    let cancelled = false;

    const getPatients = async () => {
      try {
        const res = await fetch("http://localhost:5000/getPatients/", {
          method: "POST",
          headers: { jwt_token: localStorage.token }
        });
        const parseData = await res.json();
        console.log(parseData);
        if (!cancelled) {
          setPatients(parseData);
        }
      } catch(err) {
        console.error(err.message);
      }
    }
    getPatients();
    return () => cancelled = true; 
  }, []);

  let myPatients = [];
  for (var i = 0; i < patients.length; i++) {
    myPatients.push(<h3 key={i} className="text-info"><Link to={`/dashboard/${patients[i].user_id}`}>{patients[i].user_name}</Link></h3>)
  }

  const content = (
    <div>
      <p>Name: {name}</p>
      <p>Emailï¼š{email}</p>
      <Divider />
      <Row>
        <Col offset={5}>
          <Button danger onClick={e => logout(e)}>logout</Button>
        </Col>
      </Row>
    </div>
  );

  return (
    <Layout style={{ minHeight: '100vh' }}>
      <Header className="header">
        <div className="logo" />
        <Row>
          <Col offset={22}>
            <Popover content={content} title="Your Proflie" trigger="click">
              <Avatar size="large" icon={<UserOutlined />} />
            </Popover>
          </Col>
        </Row>
      </Header>
      <Layout>
        <Sider collapsible width={200} className="site-layout-background">
          <Menu defaultSelectedKeys={["1"]} mode="inline" style={{height: '100%'}}>
            <Menu.Item key="1" icon={<AreaChartOutlined o/>}>
              <Link to="/Dashboard">Dashboard</Link>
            </Menu.Item>
            <Menu.Item key="2" icon={<BellOutlined />}>
              <Link to="/Notifications">Notifications</Link>
            </Menu.Item>
            <Menu.Item key="3" icon={<ContactsOutlined />}>
              <Link to="/Contacts">Contacts</Link>
            </Menu.Item>
          </Menu>
        </Sider>
        <Layout style={{ padding: '0 24px 24px' }}>
          <Breadcrumb style={{ margin: '16px 0' }}>
          </Breadcrumb>
          <Content
            className="site-layout-background"
            style={{
              padding: 24,
              margin: 0,
              minHeight: 280,
            }}
          >
            <div>
              <h5 className="mt-5 text-success">Clinician {name} </h5>
              <h1 className="mb-5">My Patients</h1>
              <Divider />
              <SearchableDashboardTable />
              <Divider />
              <Link to="/addPatients">
                <button className="btn btn-primary mt-5 mr-5"> Add Patients </button>
              </Link>
              {/*<Linechart />*/}
            </div>
          </Content>
        </Layout>
      </Layout>
    </Layout>
  );
};

export default ClinicianDashboard;

