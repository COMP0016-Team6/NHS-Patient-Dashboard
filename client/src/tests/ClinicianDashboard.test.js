import React from "react";
import ClinicianDashboard from "../components/ClinicianDashboard";
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { clinicianWrapper } from "./util";

it ("ClinicianDashboard snapshot matches", () => {
  const tree = render(clinicianWrapper(<BrowserRouter><ClinicianDashboard /></BrowserRouter>));
  expect(tree).toMatchSnapshot();
});
