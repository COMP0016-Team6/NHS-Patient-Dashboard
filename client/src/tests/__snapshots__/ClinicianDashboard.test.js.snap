// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClinicianDashboard snapshot matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
          style="background: rgb(6, 107, 221);"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <h4
              class="makeStyles-title-2"
            >
              Welcome, Clinician
               
              Test User One
            </h4>
            <button
              class="btn btn-info"
            >
              Logout
            </button>
          </div>
        </header>
        <h1
          class="text-center mt-5 mr-5 mb-4"
        >
          My Patients
        </h1>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-3"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-true"
          >
            <div
              class="MuiFormControl-root MuiTextField-root form-control my-3"
              style="color: rgb(97, 218, 251);"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
              >
                Search By Name
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="mt-4"
            >
              <h3>
                <a
                  href="/dashboard/2"
                >
                  Test User Two
                </a>
              </h3>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-true"
          >
            <a
              href="/addPatients"
            >
              <button
                class="btn btn-primary mt-3 mr-5"
              >
                Add Patients
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        style="background: rgb(6, 107, 221);"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <h4
            class="makeStyles-title-2"
          >
            Welcome, Clinician
             
            Test User One
          </h4>
          <button
            class="btn btn-info"
          >
            Logout
          </button>
        </div>
      </header>
      <h1
        class="text-center mt-5 mr-5 mb-4"
      >
        My Patients
      </h1>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-3"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-true"
        >
          <div
            class="MuiFormControl-root MuiTextField-root form-control my-3"
            style="color: rgb(97, 218, 251);"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Search By Name
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="mt-4"
          >
            <h3>
              <a
                href="/dashboard/2"
              >
                Test User Two
              </a>
            </h3>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-true"
        >
          <a
            href="/addPatients"
          >
            <button
              class="btn btn-primary mt-3 mr-5"
            >
              Add Patients
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
